### PR TITLE
Figure.grdimage: Let the img_out parameter work as documented

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

grdimage `-A` option works diffenrently for GMT CLI and external wrappers.

For GMT CLI, the syntax is like `gmt grdimage ingrd.nc -Aimg_out.xxx`.
For external wrappers, the syntax is like `gmt grdimage ingrid.nc -A > img_out.xxx`.

This PR makes `img_out` parameter working as documented, i.e., `img_out="out.tiff"` works.

Address https://github.com/GenericMappingTools/pygmt/issues/2599.

### CODE REVIEW

**Comment by seisman** [comment]
This PR works well for me locally on macOS, but I have no idea why it fails on CI.

**Comment by weiji14** [comment]
> This PR works well for me locally on macOS, but I have no idea why it fails on CI.

Is this only for GMT 6.5? Doesn't work locally for me with GMT 6.4.0 on Linux.

File: pygmt/src/grdimage.py [hunk 1]
```diff
@@ -175,6 +175,14 @@ def grdimage(self, grid, **kwargs):
     """
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
 
+    # Special handling of -A option.
+    # For -A option, the syntax is different for GMT CLI and external wrappers.
+    # For GMT CLI, "gmt grdimage ingrid.nc -Aimg_out;xxx".
```
Comment (yvonnefroehlich) [inline pygmt/src/grdimage.py:180]:
```suggestion
    # For GMT CLI, "gmt grdimage ingrid.nc -Aimg_out.xxx".
```

**Comment by seisman** [comment]
Ping @GenericMappingTools/pygmt-maintainers 

Please see #2599 for the related discussions, especially comments like https://github.com/GenericMappingTools/pygmt/issues/2599#issuecomment-1751795014 and https://github.com/GenericMappingTools/pygmt/issues/2599#issuecomment-1754938986.

The grdimage module usually writes PostScript codes to a hidden PostScript file, but with `-A` option, it produces a raster/vector image without writing any PS codes. So, it's a totally different workflow.

As mentioned in https://github.com/GenericMappingTools/pygmt/issues/2599#issuecomment-1754938986, `grdimage -A` works like `gdal_translate`, but allows grid projection, shading and maybe more. So, the feature is still useful.

So, I propose to:

1. Do NOT support `-A` option in the `grdimage` method
2. Fully remove `img_out` parameter from the `grdimage` method
3. If such a feature is requested by users, then we can add a new function (e.g., `pygmt.grd2img`) which wraps the `grdimage -A` option.

Thoughts?

**Comment by weiji14** [comment]
> Fully remove `img_out` parameter from the `grdimage` method

Sound good. So we're removing the alias without a deprecation cycle right? Since it didn't really work before?

**Comment by seisman** [comment]
> > Fully remove `img_out` parameter from the `grdimage` method
> 
> Sound good. So we're removing the alias without a deprecation cycle right? Since it didn't really work before?

Yes, but I think it's better to raise a warning/error if `-A`/`img_out` is used.

**Comment by seisman** [comment]
Superseded by #2907. Closing.

**Pull request closed without merging**